File: week2/atakantekatan/20. Valid Parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = collections.deque()
        val = {")":"(", "]":"[", "}":"{"}
        for p in s:
            if p in val.values():
                stack.append(p)
            elif p in val and len(stack) == 0:
                return False
            elif p in val:
                if val[p] != stack.pop():
                    return False
        return True if len(stack) == 0 else False

File: week2/Eren-Taskiran/20-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> st = new Stack();
        for(char e:s.toCharArray()){
            if(e=='(' || e=='[' || e=='{') st.add(e);
            else if(!st.isEmpty()){
                char curr = st.pop();
                if( (curr=='(' && e!=')') ||
                    (curr=='[' && e!=']') ||
                    (curr=='{' && e!='}')) return false;
            }else return false;
        }
        return st.isEmpty();
    }
}